Rubberduck.CodeAnalysis/Inspections/Concrete/ExcessiveInterfaceMembersInspection

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Linq;
+using Rubberduck.CodeAnalysis.Inspections.Abstract;
+using Rubberduck.Common;
+using Rubberduck.Parsing.Annotations;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.Resources.Inspections;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete {
+
+    internal sealed class ExcessiveInterfaceMembersInspection : DeclarationInspectionBase {
+
+        public ExcessiveInterfaceMembersInspection(IDeclarationFinderProvider declarationFinderProvider, params DeclarationType[] relevantDeclarationTypes) 
+            : base(declarationFinderProvider, relevantDeclarationTypes) {
+        }
+
+        public ExcessiveInterfaceMembersInspection(IDeclarationFinderProvider declarationFinderProvider, DeclarationType[] relevantDeclarationTypes, DeclarationType[] excludeDeclarationTypes) 
+            : base(declarationFinderProvider, relevantDeclarationTypes, excludeDeclarationTypes) {
+        }
+
+        protected override bool IsResultDeclaration(Declaration declaration, DeclarationFinder finder) {
+
+            if (IsInterfaceDeclaration(declaration)) {
+                var module = (ClassModuleDeclaration)declaration;
+                var pub = module.Members.Where(member => member.Accessibility == Accessibility.Public); //note: unsure if this handles property get/let/set correctly - doesnt account for redundancy and what needs to be counted
+                return pub.Count() > 10; //note: should be setting rather than hardcoded if desired
+            }
+            else {
+                return false;
+            }
+        }
+
+        private static bool IsInterfaceDeclaration(Declaration declaration) {//note: this is copied directly from ImplementedInterfaceMemberInspection - could this method be added to Declaration?
```

Comment:
That certainly does not belong on `Declaration`. The derived class `ClassModuleDeclaration` is the correct place. Please have a look there. I think the annotation part became part is the `IsInterface` member at some point in time, but I might be wrong.

Reply:
By rearranging the caller of this to `if(declaration is ClassModuleDeclaration module &&...) ` you can make the parameter a `ClassModule`. However, then you might just inline the function as well.

Reply:
You are correct about the annotation part from what I can tell.